frontend/src/components/widgets/CameraImageInput/CameraImageInput.tsx

```diff
@@ -0,0 +1,499 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react"
+import axios from "axios"
+import _ from "lodash"
+
+import {
+  FileUploader as FileUploaderProto,
+  FileUploaderState as FileUploaderStateProto,
+  UploadedFileInfo as UploadedFileInfoProto,
+  CameraImageInput as CameraImageInputProto,
+} from "src/autogen/proto"
+
+import Webcam from "react-webcam"
+
+import { FileSize, getSizeDisplay, sizeConverter } from "src/lib/FileHelper"
+import { FormClearHelper } from "src/components/widgets/Form"
+import { FileUploadClient } from "src/lib/FileUploadClient"
+import { WidgetStateManager, Source } from "src/lib/WidgetStateManager"
+import UIButton, { Kind } from "src/components/shared/Button"
+import { UploadFileInfo, UploadedStatus } from "../FileUploader/UploadFileInfo"
+
+export interface Props {
+  element: CameraImageInputProto
+  widgetMgr: WidgetStateManager
+  uploadClient: FileUploadClient
+  disabled: boolean
+}
+
+type FileUploaderStatus =
+  | "ready" // FileUploader can upload or delete files
+  | "updating" // at least one file is being uploaded or deleted
+
+interface State {
+  imgSrc: any
+  /**
+   * List of files dropped on the FileUploader by the user. This list includes
+   * rejected files that will not be updated.
+   */
+  files: UploadFileInfo[]
+
+  /**
+   * The most recent file ID we've received from the server. This gets sent
+   * back to the server during widget update so that it clean up
+   * orphaned files. File IDs start at 1 and only ever increase, so a
+   * file with a higher ID is guaranteed to be newer than one with a lower ID.
+   */
+  newestServerFileId: number
+
+  webcamRequestState: string
+}
+
+class CameraImageInput extends React.PureComponent<Props, State> {
```

Comment:
Should this code be rewritten with hooks or is that too much work and just do it for later on classes?